src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -472,13 +475,11 @@ private static string CompressSerializedTraceData(string json)
         /// </summary>
         public IList<string> GetOrphanedSerializables()
         {
-            var result = new List<string>();
-
-            if (!beforeFirstRunSerializables.Any())
-                return result;
+            if (beforeFirstRunSerializables.Count == 0)
+                return new List<string>();
 
-            var currentSerializables = traceData.SelectMany(td => td.RecursiveGetNestedData());
-            result.AddRange(beforeFirstRunSerializables.Where(hs => !currentSerializables.Contains(hs)).ToList());
+            var currentSerializables = traceData.SelectMany(td => td.RecursiveGetNestedData()).ToHashSet();
```

Comment:
This line with the added .ToHashSet() call and the line below are the likeliest culprits of the bad performance, as they would previously keep reloading the TraceData from the CallSite object for every .Contains test